ClientTests/SearchTests.swift

```diff
@@ -93,13 +93,14 @@ class SearchTests: XCTestCase {
         webServer.addHandlerForMethod("GET", path: "/", requestClass: GCDWebServerRequest.self) { (request) -> GCDWebServerResponse! in
             var suggestions: [String]!
             let query = request.query["q"] as! String
-            switch query {
+            switch query.stringByRemovingPercentEncoding! {
```

Comment:
`request.query` should decode the query by default -- you were probably seeing [bug 1201094](https://bugzilla.mozilla.org/show_bug.cgi?id=1201094) here. With that fixed, I don't think you'll want this change.